firmware/arduino/chessboard/motorDriver.ino

```diff
@@ -42,40 +89,68 @@ void moveSpaces(int motor[], int spaces, int dir)
   0         1        2        3
   Step pin, Dir pin, MS1 pin, MS2 pin
 */
-void setScale(int motor[], int scale) 
+
+void moveStraight(int motor[], int startCol, int startRow, int endCol, int endRow)
 {
-  if (scale == 1)  // whole steps
-  {
-    digitalWrite(motor[2], LOW);
-    digitalWrite(motor[3], LOW);
-  }
-  else if (scale == 2)  // 1/2 steps
+  // A specific motor is passed to this function since we are only moving one here
+  
+  // How many steps per space
+  int spaces, dir, numSteps;
+  int i;
+
+  // This could be two cases, x or y movement
+  if (endRow == startRow)
   {
-    digitalWrite(motor[2], HIGH);
-    digitalWrite(motor[3], LOW);
+    // X movement
+    spaces = abs(endCol - startCol);
+    dir = (endCol > startCol) ? RIGHT : LEFT;
+    setScale(xMotor, 1);
   }
-  else if (scale == 4)  // 1/4 steps
+  else if (endCol == startCol)
   {
-    digitalWrite(motor[2], LOW);
-    digitalWrite(motor[3], HIGH);
+    // Y movement
+    spaces = abs(endRow - startRow);
+    dir = (endRow > startRow) ? UP : DOWN;
+    setScale(yMotor, 1);
   }
```

Comment:
There is an unhandled case where endRow != startRow and endCol != startCol. I know this is an assumption of the function (and I think we already talked about this in Aden's PR), but if this gets called without either of those cases addressed, the program will have a garbage value when you do `numSteps = spaces * stepsPerSpace;` since `spaces` is uninitialized, which will cause weird behavior. Can simply have an `else` condition that raises an error or returns or something.